### PR TITLE
MONIT-27533: Update OTel & gRPC dependencies

### PR DESCRIPTION
**Pull request by deosu**
- Upgrade to latest jaeger dependency for our jaeger listener
- Remove alpha BOM that imports a single constant
- Add new opentelemetry-proto dependency to get exponential histograms
- Switch pom to gRPC BOM 1.41.2

### CODE REVIEW

File: proxy/pom.xml [hunk 1]
```diff
@@ -380,33 +381,17 @@
...
-        <artifactId>grpc-protobuf</artifactId>
-        <version>1.38.1</version>
-        <scope>compile</scope>
-      </dependency>
-      <dependency>
-        <groupId>io.grpc</groupId>
-        <artifactId>grpc-stub</artifactId>
-        <version>1.38.1</version>
-        <scope>compile</scope>
+        <artifactId>grpc-bom</artifactId>
```
Comment (keep94) [inline proxy/pom.xml:384]: deosu, I may have not been clear on slack, but I was suggesting that you keep oppegard's changes and your changes in separate commits.  That way it makes it easy for others to cherry pick just your changes.

File: proxy/src/test/java/com/wavefront/agent/PushAgentTest.java [hunk 2]
```diff
@@ -72,7 +67,12 @@
 import javax.net.ssl.SSLSocketFactory;
 import javax.net.ssl.TrustManager;
 
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
```
Comment (keep94) [inline proxy/src/test/java/com/wavefront/agent/PushAgentTest.java:70]: I'm just wondering if these imports are in the correct order. They don't seem to be alphabetized.

**Review by keep94 — COMMENTED** [review]
Thanks for doing this work deosu. Just a couple of minor comments.

File: proxy/src/test/java/com/wavefront/agent/PushAgentTest.java (hunk 2 above)
Comment (deosu) [reply proxy/src/test/java/com/wavefront/agent/PushAgentTest.java:70]: Yeah, you are right. But I think it is IntelliJ that is formatting these imports. Can you check on your machine and check the order?

File: proxy/pom.xml (hunk 1 above)
Comment (deosu) [reply proxy/pom.xml:384]: Ohh, I am sorry I misunderstood it. I can close this draft and submit two different commits.

**Pull request closed without merging**